assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -125,6 +127,157 @@ export default {
+              return {
+                chatChannel: this.channel,
+                channelId: this.channel.id,
+                channelTitle: this.channel.chatable.slug,
+              };
+            }
+
+            get title() {
+              return this.channel.title;
+            }
+
+            get text() {
+              return this.channel.title;
+            }
+
+            get prefixIcon() {
+              return "hashtag";
+            }
+
+            get prefixIconColor() {
+              return this.channel.chatable.color;
+            }
+
+            get prefixIconBadge() {
+              return this.channel.chatable.read_restricted && "lock";
+            }
+
+            get suffixIcon() {
+              return this.channel.unread_count > 0 && "circle";
+            }
+
+            get suffixCssClass() {
+              return this.channel.unread_mentions > 0 ? "urgent" : "unread";
+            }
+          };
+
+          const SidebarChatSection = class extends BaseCustomSidebarSection {
+            @tracked sectionLinks = [];
+
+            constructor({ sidebar }) {
+              super(...arguments);
+
+              this.sidebar = sidebar;
+              this.chatService = container.lookup("service:chat");
+              this.currentUser = container.lookup("current-user:main");
+              this.appEvents = container.lookup("service:appEvents");
+              this.appEvents.on("chat:refresh-channels", this._refreshChannels);
+              this.appEvents.on(
+                "chat:navigated-to-full-page",
+                this._refreshChannels
+              );
+            }
+
+            willDestroy() {
+              this._super(...arguments);
+              this.appEvents.off(
+                "chat:refresh-channels",
+                this._refreshChannels
+              );
+              this.appEvents.off(
+                "chat:navigated-to-full-page",
+                this._refreshChannels
+              );
+            }
+
+            @bind
+            _refreshChannels() {
+              const newSectionLinks = [];
+              this.chatService.getChannels().then((channels) => {
+                channels.publicChannels.forEach((channel) => {
+                  const trackingState =
+                    this.currentUser.chat_channel_tracking_state[channel.id];
+                  if (channel === this.chatService.activeChannel) {
+                    channel.unread_count = 0;
+                    channel.unread_mentions = 0;
+                  } else {
+                    channel.unread_count = trackingState.unread_count;
+                    channel.unread_mentions = trackingState.unread_mentions;
+                  }
+                  newSectionLinks.push(new SidebarChatSectionLink({ channel }));
+                });
+              });
+              this.sectionLinks = newSectionLinks;
+            }
+
+            get name() {
+              return I18n.t("chat.chat_channels");
```

Comment:
This is meant to be the name of the section so that we can target it with code. As such, it is not supposed to be translated.